src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -282,22 +282,30 @@ private void checkIfGroupsExists(Course course) {
      */
     @GetMapping("/courses/for-dashboard")
     @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
-    public List<Course> getAllCoursesForDashboard(Principal principal) {
+    public List<Course> getAllCoursesForDashboard(Principal principal, @RequestParam(name = "userId", required = false) String userId) {
         long start = System.currentTimeMillis();
         log.debug("REST request to get all Courses the user has access to with exercises, participations and results");
         log.debug("/courses/for-dashboard.start");
+        Boolean usesOtherLogin = false;
         User user = userService.getUserWithGroupsAndAuthorities();
+        if (authCheckService.isAdmin() && userId != null) {
```

Comment:
1) I'm not sure if I like this solution: We would have to implement this on every endpoint that needs to work for admins that login as students. This could easily be forgotten when we add a new endpoint, leading to unexpected results.
In my opinion the admin should actually be able to login as the user without knowing the password.
2) There is no documentation on what is happening at this part of the method.
3) As mentioned above I think the variable name should be more explicit as "uses student login"